operators/src/processing/temporal_raster_aggregation.rs

```diff
@@ -0,0 +1,1074 @@
+            AggregationType::Min => {
+                if self.ignore_no_data {
+                    let spec = TemporalRasterAggregationSubQuery {
+                        fold_fn: no_data_ignoring_fold_future::<P, MinIgnoreNoDataAccFunction>,
+                        no_data_value: self.no_data_value,
+                        initial_value: P::max_value(),
+                        step: self.window,
+                    };
+
+                    let s = RasterOverlapAdapter::<'a, P, _, _>::new(
+                        &self.source,
+                        query,
+                        self.tiling_specification,
+                        ctx,
+                        spec,
+                    );
+
+                    Ok(s.boxed())
+                } else {
+                    let spec = TemporalRasterAggregationSubQuery {
+                        fold_fn: fold_future::<P, MinAccFunction>,
+                        no_data_value: self.no_data_value,
+                        initial_value: P::max_value(),
+                        step: self.window,
+                    };
+
+                    let s = RasterOverlapAdapter::<'a, P, _, _>::new(
+                        &self.source,
+                        query,
+                        self.tiling_specification,
+                        ctx,
+                        spec,
+                    );
+
+                    Ok(s.boxed())
+                }
+            }
+            AggregationType::Max => {
+                if self.ignore_no_data {
+                    let spec = TemporalRasterAggregationSubQuery {
+                        fold_fn: no_data_ignoring_fold_future::<P, MaxIgnoreNoDataAccFunction>,
+                        no_data_value: self.no_data_value,
+                        initial_value: P::min_value(),
+                        step: self.window,
+                    };
+
+                    let s = RasterOverlapAdapter::<'a, P, _, _>::new(
+                        &self.source,
+                        query,
+                        self.tiling_specification,
+                        ctx,
+                        spec,
+                    );
+
+                    Ok(s.boxed())
+                } else {
+                    let spec = TemporalRasterAggregationSubQuery {
+                        fold_fn: fold_future::<P, MaxAccFunction>,
+                        no_data_value: self.no_data_value,
+                        initial_value: P::min_value(),
+                        step: self.window,
+                    };
+
+                    let s = RasterOverlapAdapter::<'a, P, _, _>::new(
+                        &self.source,
+                        query,
+                        self.tiling_specification,
+                        ctx,
+                        spec,
+                    );
+
+                    Ok(s.boxed())
+                }
+            }
+            AggregationType::Mean => Err(error::Error::NotYetImplemented), // TODO: implement
```

Comment:
I guess this also holds for the median, so do we add both as NotYetImplemented or just omit them for now?

Reply:
087b477bfeb7dff173a5258175d9c442682cbd8c